src/black/files.py

```diff
@@ -211,15 +211,27 @@ def gen_python_files(
 
     `report` is where output about exclusions goes.
     """
+
+    def is_ignored(
```

Comment:
Doesn't seem like this needs to be a nested function.

Reply:
Using a nested function allows me to short-circuit if a match is found and break the outer loop (`for child in paths:`) with its result. Using `continue` in an inner loop just jumps to the next iteration in that loop. Using `break` might help, but how to know if there was actually a match or not (breaking because `relative_path is None` or because there was a match)? I think that using a nested function makes it easier to read/understand.

Reply:
I'm suggesting to make it a separate top-level function, wouldn't that be possible?

Reply:
Oh sure, I'll update the PR with that change